webview-ui/src/components/settings/ApiOptions.tsx

```diff
@@ -306,12 +308,24 @@ const ApiOptions = ({
 
 			{errorMessage && <ApiErrorMessage errorMessage={errorMessage} />}
 
+			{/* kilocode_change start */}
 			{selectedProvider === "kilocode" && (
 				<>
 					<div style={{ marginTop: "0px" }} className="text-sm text-vscode-descriptionForeground -mt-2">
 						You get $20 for free!
 					</div>
 
+					<VSCodeTextField
+						value={apiConfiguration?.kilocodeToken || ""}
+						type="password"
+						onInput={handleInputChange("kilocodeToken")}
+						placeholder="KiloCode API Key"
+						className="w-full">
+						<div className="flex justify-between items-center mb-1">
+							<label className="block font-medium">KiloCode API Key</label>
+						</div>
+					</VSCodeTextField>
+
```

Comment:
This field is now immediately visible for everyone, which may not be great; perhaps it's easy to accidentally modify the value and cause things to stop working?